PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SurveyPopUpDialog.java

```diff
@@ -0,0 +1,224 @@
+        addWindowListener(new WindowAdapter() {
+            public void windowClosing(WindowEvent e) {
+                onCancel();
+            }
+        });
+        // call onCancel() on ESCAPE
+        contentPane.registerKeyboardAction(e -> onCancel(), KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0), JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT);
+
+        // Add listener to intellij theme change
+        LafManager.getInstance().addLafManagerListener(this.themeListener);
+        renderUiByTheme(LafManager.getInstance().getCurrentLookAndFeel());
+    }
+
+    private void renderUiByTheme(UIManager.LookAndFeelInfo theme){
+        switch (theme.getName().toLowerCase()){
+            case "intellij":
+                this.setBackGroundColor(contentPane,Color.WHITE);
+                giveFeedbackButton.setForeground(new Color(255, 255, 255));
+                giveFeedbackButton.setBackground(new Color(0,114,198));
+                notNowButton.setForeground(new Color(255, 255, 255));
+                notNowButton.setBackground(new Color(105,105,105));
+                buttonOnHoverColor = Color.LIGHT_GRAY;
+                break;
+            default:
+                this.setBackGroundColor(contentPane,null);
+                giveFeedbackButton.setForeground(null);
+                giveFeedbackButton.setBackground(null);
+                notNowButton.setForeground(null);
+                notNowButton.setBackground(null);
+                buttonOnHoverColor = Color.WHITE;
+                break;
+        }
+    }
+
+    private void setDisposeTimer(){
+        this.addMouseListener(new MouseInputAdapter() {
+            @Override
+            public void mouseEntered(MouseEvent e) {
+                SurveyPopUpDialog.this.disposeTimer.stop();
+            }
+
+            @Override
+            public void mouseExited(MouseEvent e) {
+                SurveyPopUpDialog.this.disposeTimer.restart();
+            }
+        });
+    }
+
+    private void setButtonHoverListener(JButton button){
+        button.addMouseListener(new MouseInputAdapter() {
+            Color originForegroundColor;
+
+            @Override
+            public void mouseEntered(MouseEvent e) {
+                originForegroundColor = ((JButton)e.getSource()).getForeground();
+                super.mouseEntered(e);
+                button.setForeground(buttonOnHoverColor);
+            }
+
+            @Override
+            public void mouseExited(MouseEvent e) {
+                super.mouseExited(e);
+                button.setForeground(originForegroundColor);
+            }
+        });
+    }
+
+
+    private void setBackGroundColor(JPanel panel, Color color) {
+        panel.setBackground(color);
+        for (Component child : panel.getComponents()) {
+            if (child instanceof JPanel) {
+                setBackGroundColor((JPanel) child, color);
+            }
+        }
+    }
+
+    private ImageIcon getAzureIcon() {
```

Comment:
can it be static?

Reply:
We can refer to `com.intellij.incons.AllIcons`

Reply:
Extract to PluginUtils to getIcon with customized size